knebind/solver/solver.go

```diff
@@ -115,16 +164,36 @@ func Solve(tb *opb.Testbed, topo *tpb.Topology) (*binding.Reservation, error) {
 		addIntf(link.GetANode(), link.GetAInt())
 		addIntf(link.GetZNode(), link.GetZInt())
 	}
-
 	// Build the node => links map for topology
 	for _, l := range topo.GetLinks() {
 		srcNode := name2Node[l.GetANode()]
 		dstNode := name2Node[l.GetZNode()]
 		srcIntf := s.node2Intfs[srcNode][l.GetAInt()]
 		dstIntf := s.node2Intfs[dstNode][l.GetZInt()]
-		nLink := &nodeLink{srcNode: srcNode, srcIntf: srcIntf, dstNode: dstNode, dstIntf: dstIntf}
-		s.node2Links[srcNode][dstNode] = append(s.node2Links[srcNode][dstNode], nLink)
-		s.node2Links[dstNode][srcNode] = append(s.node2Links[dstNode][srcNode], nLink)
+		group := srcNode.Interfaces[l.GetAInt()].GetGroup()
+		if group == "" {
```

Comment:
If both the srcNode and dstNode have groups, this would appear naively to drop information that might be necessary to arrive at a correct solution. I think there is an assumption there about the port groups that is unstated. Could you document the assumption and/or rationale that allows the name of the destination group to be ignored?

Reply:
Sure, will add comment here.

Reply:
I'm still concerned about the case, both with the device links and the node links, where you have port groups at both ends and links in _both_ directions. It seems like we may lose the constraint that those link belong to the same "group". I don't see a test for this case.

Reply:
I will error out if both ends have group specified, it is unsupported for new. Will also add a testcase for this.